Dev/hardknott/grub

billpittman opened the pull request:
This change allows us to use upstream grub instead of ni/grub.  Also removes some dated TPM code.  Fixes some naming issues and recipe issues that allow grub to build without warnings.

Also this has been tested on my 9032 hardware and the target still behaves wrt the CPLD operations.

Commit:
grub*.bbappend: Rename bbappend files to match recipe names

Rename bbappend files to match recipe names since the original
recipes were not versioned.

Signed-off-by: Bill Pittman <bill.pittman@ni.com>

amstewart commented on recipes-bsp/grub/grub-nilrt.inc:
```diff
@@ -1,18 +1,14 @@
-	${NILRT_GIT}/grub.git;branch=${GRUB_BRANCH} \
-	file://x86-64-Treat-R_X86_64_PLT32-as-R_X86_64_PC32.patch \
+SRC_URI += "\
+	file://cmd-test-Add-bitwise-AND-document-the-feature.patch \
```
If any of these .patches are related to each other, we should express an ordering by prepending the file name like `0001_*`. Is that not the case?

billpittman replied:
They are only related in the sense that they're all NI commits/patches.  They don't share files and they don't depend on one another.  That said I really don't care how they're named, want me to name them 0001-* 0002-* and 0003-* arbitrarily (which lets all admit will turn into alphabetical) ??

amstewart replied:
Nope; if they're not related, then I don't see a need to enforce an order.

amstewart commented on recipes-bsp/grub/grub-efi_git.bbappend:
```diff
@@ -2,13 +2,13 @@ require grub-nilrt.inc
 CFLAGS_append += "-Wno-error"
 
-PACKAGES_prepend = "${PN}-nilrt"
+PACKAGES_prepend = "${PN}-nilrt "
```
NIT: Should this line have been a part of the previous commit?

billpittman replied:
True enough, but I also removed the measure component, and that moved the "file" into this commit.  Want me to split the two changes so they can go in appropriate commits ?

amstewart replied:
I mean... I would like you to; yes. :)

amstewart commented on recipes-bsp/grub/grub/grub-safemode.cfg:
```diff
@@ -1,9 +1,5 @@
 # NI Linux RT grub boot configuration script
```
What's with all the changes in this file?

billpittman replied:
Removing the measure.mod from grub (the commit Haris added that we've now dropped), caused the measure command to be unknown, so I removed all the references to measure (since they're related to TPM).

chaitu236 replied:
Why did we add `measure` before and why are we removing it now?

billpittman replied:
measure was for TPM implementations - my understanding on that is two fold, one the API has changed, so most/some/all of our implementation will need to change, and two we never sold it, and nobody's asking for it, so its not is the mvp.

bstreiff commented on recipes-bsp/grub/grub/add-inbit-command-to-io-module.patch:
```diff
@@ -0,0 +1,86 @@
+From 48de0b2faec205e874099e0afdc14ab70919e702 Mon Sep 17 00:00:00 2001
```
Do we need this one? The bitwise-and patch describes itself as replacing the functionality of "inbit".

billpittman replied:
Totally agree with you, however looking at the grub-safemode.cfg file, there are still calls to inbit if the cpld_test value is $e5 (not sure what $e5 means, probably the "old version" of the cpld/smbios handling code?!?!?).

amstewart reviewed (changes requested):
Looks good; pending testing on a physical system and the minor NIT above.

Commit:
grub: Fix spacing error in variable

Fix spacing issue in PACKAGES_prepend variable by adding
a trailing space.  Without the trailing space, the -nilrt
version of the package was not built and caused an error
about installing files that were not part of the package.

Signed-off-by: Bill Pittman <bill.pittman@ni.com>

Commit:
grub-nilrt.inc: Use upstream grub source not ni/grub

Switched to using upstream grub and add/dropped the necessary
patches from ni/grub.  This should allow us to keep up to date
with OE and just maintain a few NI specific patches.

This change also drops the measure.mod module from grub that was
used by TPM related software.

Signed-off-by: Bill Pittman <bill.pittman@ni.com>

billpittman commented:
Both fixed now.

amstewart commented:
Do you approve, @bstreiff ?
